packages/cloudflare/src/api/durable-objects/queue.ts

```diff
@@ -0,0 +1,177 @@
+import { error } from "@opennextjs/aws/adapters/logger.js";
+import type { QueueMessage } from "@opennextjs/aws/types/overrides";
+import {
+  FatalError,
+  IgnorableError,
+  isOpenNextError,
+  RecoverableError,
+} from "@opennextjs/aws/utils/error.js";
+import { DurableObject } from "cloudflare:workers";
+
+const MAX_REVALIDATION_BY_DURABLE_OBJECT = 5;
+const DEFAULT_REVALIDATION_TIMEOUT_MS = 10_000;
+
+interface ExtendedQueueMessage extends QueueMessage {
+  previewModeId: string;
+}
+
+export class DurableObjectQueueHandler extends DurableObject<CloudflareEnv> {
+  // Ongoing revalidations are deduped by the deduplication id
+  // Since this is running in waitUntil, we expect the durable object state to persist this during the duration of the revalidation
+  // TODO: handle incremental cache with only eventual consistency (i.e. KV or R2/D1 with the optional cache layer on top)
+  ongoingRevalidations = new Map<string, Promise<void>>();
+
+  // TODO: restore the state of the failed revalidations - Probably in the next PR where i'll add the storage
+  routeInFailedState = new Map<
+    string,
+    { msg: ExtendedQueueMessage; retryCount: number; nextAlarm: number }
```

Comment:
nit: add timestamp units

```suggestion
    { msg: ExtendedQueueMessage; retryCount: number; nextAlarmMs: number }
```